tests/test_core.py

```diff
@@ -285,3 +286,38 @@ def test_api_client_do_custom_headers(config, requests_mock):
                       })
     res = client.do("GET", "/test", headers={"test": "test"})
     assert res == {"well": "done"}
+
+
+def test_error(config, requests_mock):
+    errorJson = {
+        "message":
+        "errorMessage",
+        "details": [{
+            "type": errorInfoType,
+            "reason": "errorReason",
+            "domain": "errorDomain",
+            "metadata": {
+                "etag": "errorEtag"
+            },
+        }, {
+            "type": "wrongType",
+            "reason": "wrongReason",
+            "domain": "wrongDomain",
+            "metadata": {
+                "etag": "wrongEtag"
+            }
+        }],
+    }
+
+    client = ApiClient(config)
+    requests_mock.get("/test", json=errorJson, status_code=400, )
+    with pytest.raises(DatabricksError) as raised:
+        client.do("GET", "/test", headers={"test": "test"})
+
+    errorInfos = raised.value.get_error_info()
+    assert len(errorInfos) == 1
+    errorInfo = errorInfos[0]
+    assert errorInfo.reason == "errorReason"
+    assert errorInfo.domain == "errorDomain"
+    assert errorInfo.metadata["etag"] == "errorEtag"
+    assert errorInfo.type == errorInfoType
```

Comment:
Please use snake-case everywhere here. I believe we only use PascalCase for class names in our Python code.